homeassistant/components/efergy/sensor.py

```diff
@@ -65,99 +99,99 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
 
     dev = []
     for variable in config[CONF_MONITORED_VARIABLES]:
```

Comment:
This loop seems wrong? Why loop over both the monitored variables & sensor types and checking again for a match on line 106?

Reply:
Is there a better way to get the currency and period from the config schema?